sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -122,7 +123,19 @@ public VectorizedColumnReader(
     DictionaryPage dictionaryPage = pageReader.readDictionaryPage();
     if (dictionaryPage != null) {
       try {
-        this.dictionary = dictionaryPage.getEncoding().initDictionary(descriptor, dictionaryPage);
+        PrimitiveType primitiveType = descriptor.getPrimitiveType();
+        if (primitiveType.getOriginalType() == OriginalType.DECIMAL &&
```

Comment:
could you add some comments for this code? also why it only affects vectorized read path?

Reply:
The other changes affect the `non-dictionary` read

I will add comments